packages/api/src/feature_flag/controller.test.ts

```diff
@@ -283,3 +287,182 @@ describe("POST /feature-flag", () => {
+      })
+      .expect(400);
+  });
+
+  test("should respond with 400 status if globallyEnabled is missing", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: "description",
+      })
+      .expect(400);
+  });
+
+  test("should respond with 400 status code if globallyEnabled is not a boolean", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: "description",
+        globallyEnabled: "a string",
+      })
+      .expect(400);
+  });
+
+  test("should respond with 400 status if description is missing", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        globallyEnabled: true,
+      })
+      .expect(400);
+  });
+
+  test("should respond with 400 status code if description is not a text", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: 1,
+        globallyEnabled: true,
+      })
+      .expect(400);
+  });
+
+  test("should update the feature flag in the database", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: "a description",
+        globallyEnabled: false,
+      })
+      .expect(200);
+    const result = await db.query(
+      `SELECT
+        description,
+        globally_enabled::boolean as "globallyEnabled"
+      FROM
+        feature_flag
+      WHERE
+        id = '1bdf2da6-026b-4e8e-9b57-a86b1817be5d'`
+    );
+    expect(result.rows.length).toBe(1);
+    expect(result.rows[0]).toEqual({
+      description: "a description",
+      globallyEnabled: false,
+    });
+  });
+
+  test("should respond with 500 if the database call fails", async () => {
+    jest.spyOn(db, "sql").mockImplementation(() => {
+      throw new Error("SQL error");
+    });
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: "description",
+        globallyEnabled: true,
+      })
+      .expect(500);
+  });
+
+  test("should log the error if the database call fails", async () => {
+    const testError = new Error("SQL error");
+    jest.spyOn(db, "sql").mockRejectedValueOnce(testError);
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-a86b1817be5d")
+      .send({
+        description: "description",
+        globallyEnabled: true,
+      })
+      .expect(500);
+    expect(logger.error).toHaveBeenCalled();
+  });
+
+  test("should respond with 404 status code if feature flag does not exist", async () => {
+    await agent
+      .put("/api/v2/feature-flags/1bdf2da6-026b-4e8e-9b57-not-exist-12")
+      .send({
+        description: 1,
+        globallyEnabled: true,
+      })
+      .expect(400);
```

Comment:
Should the error code here be 404?

Reply:
(I think the test is passing because it's not actually reaching the 404 error, it's getting a 400 because the ID in the URL params is not a UUID)